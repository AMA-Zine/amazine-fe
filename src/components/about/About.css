.aboutGrid {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 1fr 1fr;
  gap: 25px;
  margin: 0 25px 0 25px;
} 

.columnLeft {
  grid-column-start: 1;
  grid-column-end: span 1;
}

.columnRight {
  grid-column-start: 2;
  grid-column-end: span 1;
}

.rowTop {
  grid-row-start: 1;
  grid-row-end: span 1;
}

.rowBottom {
  grid-row-start: 2;
  grid-row-end: span 1;
}

.bioPic {
  max-width: 250px;
  max-height: 250px;
  border-radius: 50%;
}

.ghIcon {
  background-image: '/assets/githubLogo.png';
  background-size: cover;
  background-repeat: no-repeat;
  width: 30px;
  height: 30px;
  margin-bottom: 8px;
}

.liIcon {
  background-image: '/assets/linkedin.png';
  background-size: cover;
  background-repeat: no-repeat;
  width: 30px;
  height: 30px;
  margin-top: 8px;
}
