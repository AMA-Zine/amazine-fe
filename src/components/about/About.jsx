/* eslint-disable max-len */
import React from 'react';
import styles from '../app/styles.css';
import aboutstyles from './About.css';
import { Link } from 'react-router-dom';


export default function About() {
  return (
    <div className={`${styles.headerSpace} ${styles.bottomSpace}`}>
      <hr />
      <article className={styles.bannerArticle}>
        <h1 className={styles.normalGap}>
        About Us
        </h1>
      </article>
      <hr />

      <div className={aboutstyles.aboutGrid}>
        <article className={`
          ${styles.article} 
          ${aboutstyles.rowTop} 
          ${aboutstyles.columnLeft}
        `}>
          <Link key="sarah" to="/sarah">
            <img src={'/assets/sarah-picture.jpg'} alt="portrait of Sarah" className={aboutstyles.bioPic}></img>
            <h3 className={styles.titleFont}>I'm Sarah.</h3>
          </Link>
          <p className={styles.bodyFont}>Ask Me Anything!</p>
        </article>

        <article className={`
          ${styles.article} 
          ${aboutstyles.rowBottom}
          ${aboutstyles.columnLeft}
        `}>
          <Link key="michelle" to="/michelle">
            <img src={'/assets/michelle-portrait.JPG'} alt="portrait of Michelle" className={aboutstyles.bioPic}></img>
            <h3 className={styles.titleFont}>I'm Michelle.</h3>
          </Link>
          <p className={styles.bodyFont}>Ask Me Anything!</p>
        </article>

        <article className={`
          ${styles.article} 
          ${aboutstyles.rowTop}
          ${aboutstyles.columnRight}
        `}>
          <Link key="patrick" to="/patrick">
            <img src={'/assets/patrick-picture.jpg'} alt="portrait of Patrick" className={aboutstyles.bioPic}></img>
            <h3 className={styles.titleFont}>I'm Patrick.</h3>
          </Link>
          <p className={styles.bodyFont}>Ask Me Anything!</p>
        </article>

        <article className={`
          ${styles.article} 
          ${aboutstyles.rowBottom}
          ${aboutstyles.columnRight}
        `}>
          <Link key="josh" to="/josh">
            <img src={'/assets/josh-picture.jpeg'} alt="portrait of Josh" className={aboutstyles.bioPic}></img>
            <h3 className={styles.titleFont}>I'm Josh.</h3>
          </Link>
          <p className={styles.bodyFont}>Ask Me Anything!</p>
        </article>

      </div>
      <hr />

    </div>
  );
}
