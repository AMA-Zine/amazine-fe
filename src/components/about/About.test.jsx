import React from 'react';
import { render, cleanup } from '@testing-library/react';
import About from './About';
import { MemoryRouter } from 'react-router-dom';

describe('About component', () => {
  afterEach(() => cleanup());
  it('renders About Component', () => {
    const { asFragment } = render(<MemoryRouter><About /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});
