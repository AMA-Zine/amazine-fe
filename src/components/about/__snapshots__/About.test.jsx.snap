// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component renders About Component 1`] = `
<DocumentFragment>
  <div
    class="headerSpace bottomSpace"
  >
    <hr />
    <article
      class="bannerArticle"
    >
      <h1
        class="normalGap"
      >
        About Us
      </h1>
    </article>
    <hr />
    <div
      class="aboutGrid"
    >
      <article
        class="
          article 
          rowTop 
          columnLeft
        "
      >
        <a
          href="/sarah"
        >
          <img
            alt="portrait of Sarah"
            class="bioPic"
            src="/assets/sarah-picture.jpg"
          />
          <h3
            class="titleFont"
          >
            I'm Sarah.
          </h3>
        </a>
        <p
          class="bodyFont"
        >
          Ask Me Anything!
        </p>
      </article>
      <article
        class="
          article 
          rowBottom
          columnLeft
        "
      >
        <a
          href="/michelle"
        >
          <img
            alt="portrait of Michelle"
            class="bioPic"
            src="/assets/michelle-portrait.JPG"
          />
          <h3
            class="titleFont"
          >
            I'm Michelle.
          </h3>
        </a>
        <p
          class="bodyFont"
        >
          Ask Me Anything!
        </p>
      </article>
      <article
        class="
          article 
          rowTop
          columnRight
        "
      >
        <a
          href="/patrick"
        >
          <img
            alt="portrait of Patrick"
            class="bioPic"
            src="/assets/patrick-picture.jpg"
          />
          <h3
            class="titleFont"
          >
            I'm Patrick.
          </h3>
        </a>
        <p
          class="bodyFont"
        >
          Ask Me Anything!
        </p>
      </article>
      <article
        class="
          article 
          rowBottom
          columnRight
        "
      >
        <a
          href="/josh"
        >
          <img
            alt="portrait of Josh"
            class="bioPic"
            src="/assets/josh-picture.jpeg"
          />
          <h3
            class="titleFont"
          >
            I'm Josh.
          </h3>
        </a>
        <p
          class="bodyFont"
        >
          Ask Me Anything!
        </p>
      </article>
    </div>
    <hr />
  </div>
</DocumentFragment>
`;
